File: Fibonacci/fib.scala
object Fib{
	def main(args: Array[String]){
		val n=40
		println("fib("+n+") = "+fib(n))
	}
	def fib(n: Int): Int = {
		if(n==0||n==1)
			1
		else
			fib(n-1)+fib(n-2)
	}
}

File: Fibonacci/fib.rs
fn main() {
    let n = 40;
    println!("fib({}) = {}", n, fib(n));
}

fn fib(n: u8) -> u64 {
    match n {
        0 | 1 => 1,
        n => fib(n - 1) + fib(n - 2),
    }
}